tests/kernel/mutex/mutex_api/src/test_mutex_apis.c

```diff
@@ -140,18 +282,174 @@ void test_mutex_lock_unlock(void)
 	tmutex_test_lock_unlock(&kmutex);
 }
 
+/**
+ * @brief Test recursive mutex
+ * @details To verify that getting a lock of a mutex already locked will
+ * succeed and waiters will be unblocked only when the number of locks
+ * reaches zero.
+ * @ingroup kernel_mutex_tests
+ */
+void test_mutex_recursive(void)
+{
+	k_mutex_init(&mutex);
+
+	/**TESTPOINT: when mutex has no owner, we cannot unlock it */
+	zassert_true(k_mutex_unlock(&mutex) == -EINVAL,
+			"fail: mutex has no owner");
+
+	zassert_true(k_mutex_lock(&mutex, K_NO_WAIT) == 0,
+			"Failed to lock test_mutex");
+
+	/**TESTPOINT: The current thread does own the mutex */
+	zassert_true(k_mutex_lock(&mutex, K_NO_WAIT) == 0,
+			"Failed to recursively lock mutex");
```

Comment:
The typical failure of a lock failing to be recursive is that it deadlocks.  Strictly, you should have another thread here to detect the hang and recover.  But in this case we know the code already works and just want coverage, I guess.  Timed out tests are failures too.

Reply:
Yes, It can slightly increase the code coverage. I rework it by adding a waiter thread that will try to get the mutex during recursive, but I guess I didn't get your point. Excuse me, could please tell me more detail about how to improve this? thanks very much.